### PR TITLE
Address data race in block.Stream()

### PR DESCRIPTION
**Pull request by prateek**

### CODE REVIEW

**Comment by robskillington** [comment]
Ouch.  Good find, want to add a test?

File: src/dbnode/storage/block/block.go [hunk 1]
```diff
@@ -223,6 +231,15 @@ func (b *dbBlock) Stream(blocker context.Context) (xio.BlockReader, error) {
 		return stream, nil
 	}
 
+	b.RUnlock()
+	lockUpgraded = true
```
Comment (robskillington) [inline src/dbnode/storage/block/block.go]: Btw, maybe we should rename the lowercase `stream(...)` to `streamWithRLock(...)`?

Comment (prateek) [reply src/dbnode/storage/block/block.go]: sure thing

File: src/dbnode/storage/block/block.go [hunk 2]
```diff
@@ -223,6 +231,15 @@ func (b *dbBlock) Stream(blocker context.Context) (xio.BlockReader, error) {
...
+	b.Lock()
+
+	// NB: need to re-check as we just upgraded the lock.
+	if b.mergeTarget == nil {
```
[the remark this answers is not in the document]
Comment (prateek) [reply src/dbnode/storage/block/block.go]: thanks, good catch! added in the next patch

**Pull request merged**